provider/provider.go

```diff
@@ -333,17 +335,40 @@ func (p *dockerNativeProvider) Create(ctx context.Context, req *rpc.CreateReques
 	if err != nil {
 		return nil, errors.Wrapf(err, "malformed resource inputs")
 	}
+	if req.GetPreview() {
+		// verify buildOnPreview is Known; if not, send warning and continue.
+		if inputs["buildOnPreview"].ContainsUnknowns() {
+			msg := "buildOnPreview is unresolved; cannot build on preview. Continuing without preview image build. " +
+				"To avoid this warning, set buildOnPreview explicitly, and ensure all inputs are resolved at preview."
+			err = p.host.Log(ctx, "warning", urn, msg)
+			if err != nil {
+				return nil, err
+			}
+			return &rpc.CreateResponse{
+				Properties: req.GetProperties(),
+			}, nil
+		}
+		// if we're in preview mode and buildOnPreview is set to false, we return the inputs
+		if !inputs["buildOnPreview"].BoolValue() {
+			return &rpc.CreateResponse{
+				Properties: req.GetProperties(),
+			}, nil
+		}
+		// buildOnPreview needs all inputs to be resolved. Warn and continue without building the image
+		// TODO: there is room for some future granularity here - we should be able to build a local image without
+		// (TODO cont) knowing inputs for a registry, for example.
+		if inputs["buildOnPreview"].BoolValue() && inputs.ContainsUnknowns() {
```

Comment:
We already know here that `inputs["buildOnPreview"].BoolValue()` is `true`, since if it wasn't we would have exited the function on L352.
```suggestion
		if inputs.ContainsUnknowns() {
```